### ISSUE TITLE
Can this now be achieved with the Maker channel?

### ISSUE THREAD

**Issue by codecowboy** [kaplas/ifttt-channel-extensions#1]
https://ifttt.com/maker

**Comment by kaplas** [comment]
Totally! Thanx for the info @codecowboy ; I'll update the readme right away :)

**Comment by codecowboy** [comment]
Thanks. However, I'm not actually sure if the Maker channel will do what I want. I want to be able to pass details of a tweet to the Maker channel and use it in a webhook and I'm not sure if that is possible?

**Comment by jamesmstone** [comment]
could you use the twitter channel https://ifttt.com/twitter

**Comment by codecowboy** [comment]
Yes, the twitter channel will be the if THIS part. The Maker channel would be the THEN that part. I want to pass details of the tweet to a webhook.

**Comment by jamesmstone** [comment]
@codecowboy  Just so I'm on the same page do you mean something like this .... ![image](https://cloud.githubusercontent.com/assets/4206501/10879955/5c43b2f8-81ab-11e5-9f08-02fa1a88cc6a.png)
 ![image](https://cloud.githubusercontent.com/assets/4206501/10879682/8d56a99c-81a9-11e5-91d4-0939f0ab84e9.png) Sorry IFTTT doesn't allow sharing of maker channel recipes

**Comment by kaplas** [comment]
Sorry everyone, I think I closed this issue a little bit too hastily yesterday. I'm reopening this one for further discussion.

I would say that there are three integral parts of the problem of "extending IFTTT":
1. Is it possible to send an event from IFTTT?
2. What format is used to transport the event?
3. Who receives and handles the event?

Previously, with `ifttt-channel-extensions` the answers were:
1. Yes, via a WordPress hack
2. An XML RPC payload (cumbersome)
3. `ifttt-channel-extensions` handles actions via custom PHP script

Now, with the Maker Channel, the situation is
1. Yes, via the Maker Channel
2. Standard JSON webhook
3. ???

So in effect, the Maker Channel does not solely replace this project. To fill the gap, I have some very interesting ideas on how to make "an IFTTT extension webserver" quite easily. I try to manage a couple of days for this experiment in the near future. An interesting thing to note is that the Maker Channel makes it possible to also signal IFTTT from the server and not just the other way around. I think that makes things two times more interesting.

In the mean time, you may want to look into for example [webtask.io](https://webtask.io/), which makes it possible to execute tasks on webhook calls. However, that may be a bit pricy, depending on what you are trying to achieve.

@codecowboy Could you share a bit more details what is that you are trying to achieve. Now this sounds a bit too much of an [XY problem](http://xyproblem.info/) to me, so that neither me or @jamesmstone can really help you enough :)

@jamesmstone I'm glad [you asked](https://github.com/kaplas/ifttt-channel-extensions/commit/413be4ef10192cdb7a8f0fb5d04f6311202b9550#commitcomment-14122642) on how you can stay tuned :) People like you really give me motivation to return my old projects as well! You'll be the first one to hear about new stuff by
- [Subscribing to my email list](http://kaplas.net/subscribe) or
- Following me [on Twitter](https://twitter.com/kaplas) or [on GitHub](https://github.com/kaplas) or
- Starring [`ifttt-extensions`](https://github.com/kaplas/ifttt-extensions), the placeholder project I just created

I will also post to this thread as soon as I have something to show to you.

**Comment by codecowboy** [comment]
@jamesmstone @kabias here's what I want to do:

When a user tweets something (possibly other actions too), post the tweet and image to a NodeBB forum
NodeBB has a write API.
the API requires authentication either via an HTTP header or JWT (JSON Web Token)
The Maker Channel can't generate JWTs and doesn't allow you to alter the header of a web request.

So that's why I ended up here. I could just write my own script and not involve IFTTT at all i.e do all the Twitter API calls myself. Or I can write a script which receives the web request from IFTTT, adds the required authentication and passes it to NodeBB. At some point (soon), it will just be easier / less hassle to code this by hand.

https://webtask.io looks interesting but I doubt this particular client will want to pay $9 / month. Someone also mentioned hookify.io to me.
